### PR TITLE
[FLASK] Update UI (for audit) (UI facelift)

### LINKED ISSUE #19352: Design audit: Custom dialogs, success, and failed screens
**_Custom dialogs_, _success_, and _failed_ screens suggestions:**
- [ ] New fixed (always visible) header-like authorship component with `md` drop-shadow
- [ ] Status icon above success / failed
- [ ] Capitalized 'OK'
- [ ] Fixed (always visible) footer container with md drop-shadow

### PR DESCRIPTION
**Pull request by david0xd**
## Explanation
Fixes https://github.com/MetaMask/metamask-extension/issues/19352

This PR is addressing changes requested in the ticket above.
Here are the Figma designs:

Custom dialogs: https://www.figma.com/file/Wlwk0cTx6PQkNhiX7vN13e/Snaps-Platform-v1?type=design&node-id=1%3A11809&t=kYhmv6mIpKd3I5mV-1
Status screens: https://www.figma.com/file/Wlwk0cTx6PQkNhiX7vN13e/Snaps-Platform-v1?type=design&node-id=1%3A10182&t=kYhmv6mIpKd3I5mV-1

### After
Screenshot 2023-06-02 at 17 07 46
Screenshot 2023-06-02 at 17 54 52
Screenshot 2023-06-02 at 18 25 29
Screenshot 2023-06-02 at 18 27 26
Screenshot 2023-06-05 at 14 56 33
Screenshot 2023-06-05 at 15 02 04

### Before
Screenshot 2023-06-02 at 18 53 26
Screenshot 2023-06-02 at 18 53 09
Screenshot 2023-06-02 at 18 52 50

## Manual Testing Steps
Go through the following flows and confirm that changes are correct:
- Install flow
- Snap custom UI (snap confirmation / dialog)

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: test/e2e/snaps/test-snap-update.spec.js [hunk 1]
```diff
@@ -72,10 +72,10 @@ describe('Test Snap update', function () {
           tag: 'button',
         });
 
-        await driver.waitForSelector({ text: 'Ok' });
+        await driver.waitForSelector({ text: 'OK' });
 
         await driver.clickElement({
-          text: 'Ok',
+          text: 'OK',
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-update.spec.js:78]: This file has more "OK"s to correct 😅

File: app/_locales/en/messages.json [hunk 2]
```diff
@@ -3807,14 +3804,22 @@
     "message": "Grant $2 public key access to $1",
     "description": "The first parameter is the name of the snap and the second one is the protocol"
   },
+  "snapInstallationErrorDescription": {
+    "message": "$1 couldn’t be installed.",
+    "description": "Error description used when snap installation fails."
```
Comment (FrederikBolding) [inline app/_locales/en/messages.json:3809]:
```suggestion
    "description": "Error description used when snap installation fails. $1 is the snap name."
```

File: ui/components/app/snaps/snap-ui-renderer/index.scss [hunk 3]
```diff
@@ -15,6 +15,14 @@
     height: 100%;
   }
 
+  &__panel > *:first-child {
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-renderer/index.scss:18]: Why is this needed now?

File: ui/pages/permissions-connect/snaps/snap-result/snap-result.js [hunk 4]
```diff
@@ -74,22 +83,43 @@ export default function SnapResult({
...
+                </Text>,
+              ])}
             </Text>
           </Box>
         )}
         {hasError && (
           <InstallError
             error={requestState.error}
-            title={t('snapResultError')}
+            title={t('snapInstallationErrorTitle')}
```
Comment (FrederikBolding) [inline ui/pages/permissions-connect/snaps/snap-result/snap-result.js:116]: We cannot assume that `SnapResult` is only used for snap install, it is also used for snap update, so we need to change this copy dynamically based on the previous flow. Should be available as `requestState.type`

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: ui/pages/permissions-connect/snaps/snap-result/snap-result.js (hunk 4 above)
Comment (david0xd) [reply ui/pages/permissions-connect/snaps/snap-result/snap-result.js]: Added changes which will leave update flow untouched. So, only the install flow will be affected. I think update flow is supposed to be updated in another ticket?

File: app/_locales/en/messages.json (hunk 2 above)
Comment (david0xd) [reply app/_locales/en/messages.json]: Updated.

File: ui/pages/permissions-connect/snaps/snap-result/snap-result.js (hunk 4 above)
Comment (FrederikBolding) [reply ui/pages/permissions-connect/snaps/snap-result/snap-result.js]: I have made other changes to the update flow in another PR, but not to the result screens. Do you mind including the updated result screens for updates in this PR as well? Should mostly be copy given that it leverages most of what you've changed already.

Comment (david0xd) [reply ui/pages/permissions-connect/snaps/snap-result/snap-result.js]: Added new messages and design for update failed and success screens.

File: ui/components/app/snaps/snap-ui-renderer/index.scss (hunk 3 above)
Comment (FrederikBolding) [reply ui/components/app/snaps/snap-ui-renderer/index.scss:18]: Won't this rule also match nested panels? So nested panels first and last element will also have no margins?

File: app/_locales/en/messages.json [hunk 5]
```diff
@@ -3804,22 +3804,41 @@
...
+  "snapUpdateErrorTitle": {
+    "message": "Update failed",
+    "description": "Error title used when snap update fails."
+  },
   "snapUpdateRequest": {
     "message": "$1 wants to update $2 to $3 which gives it the following permissions. Only continue if you trust $2.",
     "description": "$1 is the dApp origin requesting the snap, $2 is the snap name and $3 is the snap version."
   },
+  "snapUpdateSuccess": {
+    "message": "Update complete "
```
Comment (FrederikBolding) [inline app/_locales/en/messages.json:3840]:
```suggestion
    "message": "Update complete"
```

Comment (david0xd) [reply app/_locales/en/messages.json]: Fixed.

File: ui/components/app/snaps/snap-ui-renderer/index.scss (hunk 3 above)
Comment (FrederikBolding) [reply ui/components/app/snaps/snap-ui-renderer/index.scss:18]: The existing code is meant to do this already too? 🤔  https://github.com/MetaMask/metamask-extension/blob/7420ea9a5dca41dd2b74c9d19c88045c9588d8fd/ui/components/app/snaps/snap-ui-renderer/index.scss#L30-L40

**Review by FrederikBolding — DISMISSED** [review]

File: ui/pages/permissions-connect/snaps/snap-result/snap-result.js [hunk 6]
```diff
@@ -34,11 +43,43 @@ export default function SnapResult({
...
+      </Text>,
+    ]);
+  } else if (requestState.type === 'wallet_updateSnap') {
+    failedScreenTitle = t('snapUpdateErrorTitle');
+    failedScreenDescription = t('snapUpdateErrorDescription', [
+      <Text as={ValidTag.Span} key="1" fontWeight={FontWeight.Medium}>
+        {snapName}
+      </Text>,
+    ]);
+  }
```
Comment (Mrtenz) [inline ui/pages/permissions-connect/snaps/snap-result/snap-result.js:81]: I feel like this could be cleaner if it was moved into a separate function or something, which just returns the React component to render.

```ts
const SnapResult = ({ isLoading, hasError }) => {
  function getResult() {
    if (isLoading) {
      return ...; // Loading components
    }

    if (hasError) {
      const title = requestState.type === 'wallet_installSnap' ?
        ? t('snapInstallationErrorTitle')
        : t('snapUpdateErrorDescription');

      // etc.

      return ...; // Error components
    }

    // etc.
  }
};
```

This avoids having a bunch of mutable variables and if-else-statements.

**Review by Mrtenz — DISMISSED** [review]

File: ui/pages/permissions-connect/snaps/snap-result/snap-result.js (hunk 6 above)
Comment (david0xd) [reply ui/pages/permissions-connect/snaps/snap-result/snap-result.js]: There are couple of problems with it. Title and description are different things and each can have different content for update, install and default scenario. The best approach would be to create separate components for these and have them render what's needed, but this ticket was supposed to be only the UI facelift. Also, I think other members are working on some parts of this flow, so I would like to avoid having something like that at this moment. This should be refactored in a different ticket I think.
The usage of nested ternary is disallowed in this repository and ESLint will complain `ESLint: Do not nest ternary expressions.(no-nested-ternary)`. I would like to respect that.

I did some refactoring to reduce complexity around bunch of mutable variables all in one scope and compacted these parts of a template into functions that would make rendering content based on input. It should be more clean now I think.

**Review by FrederikBolding — APPROVED** [review]

File: ui/pages/permissions-connect/snaps/snap-result/snap-result.js (hunk 6 above)
Comment (FrederikBolding) [reply ui/pages/permissions-connect/snaps/snap-result/snap-result.js]: Yeah, I think this needs a refactor, but this seems to be a decently readable. What do you think @Mrtenz ?

**Review by Mrtenz — APPROVED** [review]
